apps/server-discover/main.c

```diff
@@ -262,6 +299,37 @@ int main(int argc, char *argv[])
             BACNET_MAX_INSTANCE);
         return 1;
     }
+    if (specific_address) {
+        bacnet_address_init(&dest, &mac, dnet, &adr);
+        if (adr.len && mac.len) {
```

Comment:
It seems that apps/readprop has some cruft laying around.  The bacnet_address_init() does (mostly) everything that the set of if/else is doing. [apps/writeprop](https://github.com/bacnet-stack/bacnet-stack/blob/31af2507fbe201029d438cb06d84c4e6d364ebed/apps/writeprop/main.c#L457) and a few other apps have been updated. 

Here is what this section should look like:
```
        if (specific_address) {
            bacnet_address_init(&dest, &mac, dnet, &adr);
        }
```